Website - `CodeBlock` height toggle docs (#2845)
hashicorp/design-system: website/docs/components/code-block/partials/guidelines/guidelines.md

In the Figma component, the code examples have the appropriate number of lines by default but must be manually hidden or shown to match the length of custom snippets.
!!!

## Height toggle button

> When displaying longer code snippets, it can be helpful to let users expand and collapse the content. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button appears at the bottom of the block, allowing users to expand it. Interacting with the button again collapses it to it's original height.

Reviewer: To me, this text kind of makes it sound like displaying the height toggle button is an optional feature when, actually, we implemented it to always be included if a `maxHeight` is set and the content overflows at the max-height.

Reviewer 2: @KristinLBradley Ah, I can change the language to read less like it's optional.

> When displaying longer code snippets, it can be helpful to let users expand and collapse the content. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button appears at the bottom of the block, allowing users to expand it. Interacting with the button again collapses it to it's original height.

Reviewer 2: ```suggestion
When displaying longer code snippets, it can be helpful to let users expand and collapse the content. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button automatically appears at the bottom of the block, allowing users to expand it. Clicking the button again collapses the block back to its original height.
```

> When displaying longer code snippets, it can be helpful to let users expand and collapse the content. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button automatically appears at the bottom of the block, allowing users to expand it. Clicking the button again collapses the block back to its original height.

Reviewer 3: Couple of suggestions here:
- Consistent use of "interacting" vs clicking
- Might be helpful to say _why_ it's helpful to the user to expand/collapse the component
- "…will be displayed…" is a bit more formal/active

```suggestion
When displaying longer code snippets, it can be helpful to let users expand and collapse the content to save space in the UI. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button will be displayed at the bottom of the block, allowing users to expand it. Interacting with the button again collapses the block back to its original height.
```

> When displaying longer code snippets, it can be helpful to let users expand and collapse the content to save space in the UI. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button will be displayed at the bottom of the block, allowing users to expand it. Interacting with the button again collapses the block back to its original height.

Reviewer: Working on tweaking this as it still seems a bit awkward... (I'll make a suggested edit)

> When displaying longer code snippets, it can be helpful to let users expand and collapse the content to save space in the UI. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button will be displayed at the bottom of the block, allowing users to expand it. Interacting with the button again collapses the block back to its original height.

Reviewer: ```suggestion
For longer code content, it can be helpful to set a `maxHeight` for the Code Block to limit how much code is shown by default within the UI layout. If the content exceeds this height, a “Show more code” button will be displayed at the bottom of the Code Block, allowing users to expand it. Activating this button again will collapse the content back to its original height.
```

Reviewer: @andgen404 How's this? (Above suggested edit)

In the Figma component, the code examples have the appropriate number of lines by default but must be manually hidden or shown to match the length of custom snippets.
!!!

## Height toggle button

> Use the height toggle button to let users expand and collapse longer code snippets.

Reviewer 3: [suggestion] I might suggest combining these two sentences into a paragraph and maybe flip flop the order. WDYT of something like this:

When a `maxHeight` is set on the Code Block and the code content height exceeds this value, a button to toggle the height of the Code Block will be displayed. In Figma this is enabled by setting `hasToggleHeightButton` to true and can be used to let expand and collapse longer code snippets.

This might more closely ties the Figma property to height concept and brings the sentences together into a single paragraph.

Reviewer 2: Ah, yeah, that makes sense to me. I like tying the figma prop closer to the height behavior. Thanks! I'll try it out.

> Use the height toggle button to let users expand and collapse longer code snippets.
> 
> When `maxHeight` is set and the code content is taller than that value, a **Show more code** button appears at the bottom of the Code Block.

Reviewer 2: ```suggestion
When displaying longer code snippets, it can be helpful to let users expand and collapse the content. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button appears at the bottom of the block, allowing users to expand it. Clicking again collapses it back.

In Figma, this is enabled by setting `hasToggleHeightButton` to true.

The button is placed inside a footer element that only appears when the content overflows, keeping the layout clean when the toggle isn’t needed.
```

![Collapsed Code Block showing limited lines and a 'Show more code' button at the bottom.](/assets/components/code-block/code-block-collapsed.png)

> - Clicking this button removes the height limit, allowing the full code snippet to be displayed. 

Reviewer 3: I wouldn't bullet these out since they are single items.

```suggestion
Interacting with this button removes the height limit, expanding the Code Block to display the full code snippet. 
```

![Expanded Code Block with all lines visible and a 'Show less code' button displayed at the bottom](/assets/components/code-block/code-block-expanded.png)

> - Clicking it again collapses the Code Block back to its original height.

Reviewer 3: ```suggestion
Interacting with it again collapses the Code Block back to its set `maxHeight`.
```

> The button is placed inside a footer element that only appears when `maxHeight` is set and the content overflows. This keeps the layout clean when the toggle isn’t needed.

Reviewer 3: Reading this now, it might make sense to try to combine this with the introductory paragraph above.

Reviewer 2: Combined!

Reviewer 3: This can be removed now
```suggestion
```

In the Figma component, the code examples have the appropriate number of lines by default but must be manually hidden or shown to match the length of custom snippets.
!!!

## Height toggle button

> When displaying longer code snippets, it can be helpful to let users expand and collapse the content. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button appears at the bottom of the block, allowing users to expand it. Clicking again collapses it back.

Reviewer 3: In some of my other suggestions I noted using "interacting" vs "clicking" which is more inclusive, so noting that here.

```suggestion
When displaying longer code snippets, it can be helpful to let users expand and collapse the content. Setting a `maxHeight` on the Code Block limits how much is shown by default. If the content exceeds this height, a **Show more code** button appears at the bottom of the block, allowing users to expand it. Interacting with the button again collapses it to it's original height.
```

In Figma, this is enabled by setting `hasToggleHeightButton` to true.

> 

Reviewer 2: ```suggestion
```

> The button is placed inside a footer element that only appears when the content overflows, keeping the layout clean when the toggle isn’t needed.

Reviewer 3: I would combine this with the previous sentence to create a paragraph.
